node/src/components/linear_chain.rs

```diff
@@ -58,7 +52,7 @@ pub enum Event<I> {
     KnownLinearChainBlock(Box<Block>),
     /// Finality signature received.
     /// Not necessarily _new_ finality signature.
-    FinalitySignatureReceived(Box<FinalitySignature>),
+    FinalitySignatureReceived(Box<FinalitySignature>, bool),
```

Comment:
Now that I think of it, I'd actually make this two events: `FinalitySignatureReceived` and `FinalitySignatureCreated`. But we can do that later, in another PR.

Reply:
…or `FinalitySignatureReceived(Signature)` with the enum you added below?

Reply:
I tried but this is actually a lot of work.